interfaces.go

```diff
@@ -5,14 +5,31 @@
 package optimize
 
 // A Method can optimize an objective function.
+//
+// It uses a reverse-communication interface design represented by the Init()
+// and Iterate() methods to pass and retrieve information from the client. Its
```

Comment:
"methods to receive information about the objective function". It doesn't pass information to the client aside from X location.